pkgs/development/python-modules/uproot/default.nix

```diff
@@ -0,0 +1,27 @@
+{lib, fetchFromGitHub, buildPythonPackage, numpy }:
+
+buildPythonPackage rec {
+  pname = "uproot";
+  version = "2.8.16";
+  src = fetchFromGitHub {
+    owner  = "scikit-hep";
+    repo   = "uproot";
+    rev    = "${version}";
+    sha256 = "07f6h7zcvjc4ynqydp12cbq8c6bb8964s1i65cc25py4mg46f4my";
+  };
+
+  # Why does it propagate packages that are used for testing?
```

Comment:
numpy is listed in install_requires, you're sure it isn't a runtime dependency?

Reply:
You can't do anything sensible in uproot without Numpy. It's an absolute requirement. I hope the setup.py reflects that.

Many of the other libraries that uproot might use are optional, particularly compression codecs that aren't found in all ROOT files.

Reply:
Yes, setup.py reflects this. But the comment

> Why does it propagate packages that are used for testing?

doesn't make sense in that case.